shell/gpu/gpu_surface_software.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/gpu/gpu_surface_software.h"
+
+#include <memory>
+#include "lib/ftl/logging.h"
+
+namespace shell {
+
+GPUSurfaceSoftware::GPUSurfaceSoftware(GPUSurfaceSoftwareDelegate* delegate)
+    : delegate_(delegate), weak_factory_(this) {}
+
+GPUSurfaceSoftware::~GPUSurfaceSoftware() = default;
+
+bool GPUSurfaceSoftware::Setup() {
+  return IsValid();
+}
+
+bool GPUSurfaceSoftware::IsValid() {
+  return delegate_ != nullptr;
+}
+
+std::unique_ptr<SurfaceFrame> GPUSurfaceSoftware::AcquireFrame(
+    const SkISize& size) {
+  if (!IsValid()) {
+    return nullptr;
+  }
+
+  auto backing_store = delegate_->AcquireBackingStore(size);
+
+  if (backing_store == nullptr) {
+    return nullptr;
+  }
+
+  if (size != SkISize::Make(backing_store->width(), backing_store->height())) {
+    return nullptr;
+  }
+
+  SurfaceFrame::SubmitCallback on_submit =
+      [
+        self = weak_factory_.GetWeakPtr(),
+        original_backing_store = backing_store
```

Comment:
Hmm. I am not sure why clang didn't warn me about this unused capture. Will fix and try to add flag to clang. I know it is supported. Thought it was enabled too.